### PR TITLE
Bugfixes

### LINKED ISSUE #3496: client disconnects after few seconds with "got message barrier" error
# Required information

 * Distribution: Ubuntu 17.04
 * The output of "lxc info" or if that fails:
```
  driver: lxc
  driver_version: 2.0.8
  kernel: Linux
  kernel_architecture: x86_64
  kernel_version: 4.10.0-28-generic
  server: lxd
  server_pid: 2209
  server_version: "2.12"
  storage: dir
  storage_version: "1"
```

# Issue description

Latest master version of `lxc` client breaks connection after 5 or so seconds with `←[36mDBUG←[0m[07-05|16:52:09] got message barrier` error.

# Steps to reproduce

 1. `lxc exec c1 --debug -- bash`
 2. press enter few times
 3. see the error

# Information to attach

 - [ ] any relevant kernel output (dmesg)
 - [ ] container log (lxc info NAME --show-log)
 - [ ] main daemon log (/var/log/lxd.log)
 - [x] output of the client with --debug
```
...
←[36mDBUG←[0m[07-05|16:55:20] Connected to the websocket
←[36mDBUG←[0m[07-05|16:55:20] Window size is now: 182x67
←[36mDBUG←[0m[07-05|16:55:20] Connected to the websocket
←[32mINFO←[0m[07-05|16:55:20] Sending request to LXD                   ←[32metag←[0m= ←[32mmethod←[0m=GET ←[32murl←[0m=https://192.1...
# ←[36mDBUG←[0m[07-05|16:55:20] Got response struct from LXD
←[36mDBUG←[0m[07-05|16:55:20]
        {
                "id": "596057f4-83a8-46b0-836d-b8f7d4df50b4",
                "class": "websocket",
                "created_at": "2017-07-05T16:55:25.854034643+03:00",
                "updated_at": "2017-07-05T16:55:25.854034643+03:00",
                "status": "Running",
                "status_code": 103,
                "resources": {
                        "containers": [
                                "/1.0/containers/c1"
                        ]
                },
                "metadata": {
                        "fds": {
                                "0": "e17077569d9fa5f9b0a601fa733602fb0180c636d897ad46e46120d9399d7741",
                                "control": "34e23b4d9dd6f6db09a63aae2a8316522c1a8ea2af6b0124f38c3bed777f8a68"
                        }
                },
                "may_cancel": false,
                "err": ""
        }

# ls -la
total 24
drwx------  3 root root 4096 May 30 10:31 .
drwxr-xr-x 22 root root 4096 Apr 12 09:01 ..
-rw-------  1 root root  264 May 30 11:16 .bash_history
-rw-r--r--  1 root root 3106 Oct 22  2015 .bashrc
-rw-r--r--  1 root root  148 Aug 17  2015 .profile
drwx------  2 root root 4096 May 23 16:05 .ssh
# l←[36mDBUG←[0m[07-05|16:55:25] got message barrier
```
 - [ ] output of the daemon with --debug

### PR DESCRIPTION
**Pull request by stgraber**
Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>

### CODE REVIEW

**Comment by techtonik** [comment]
I wonder if it is possible to add end2end test with remote LXD server for AppVeyor?

File: lxc/exec_windows.go [hunk 1]
```diff
@@ -32,11 +34,24 @@ func (c *execCmd) getTERM() (string, bool) {
...
 
 func (c *execCmd) controlSocketHandler(control *websocket.Conn) {
-	// TODO: figure out what the equivalent of signal.SIGWINCH is on
-	// windows and use that; for now if you resize your terminal it just
-	// won't work quite correctly.
-	err := c.sendTermSize(control)
-	if err != nil {
-		logger.Debugf("error setting term size %s", err)
+	ch := make(chan os.Signal, 10)
+	signal.Notify(ch, os.Interrupt)
```
Comment (techtonik) [inline lxc/exec_windows.go]: Make a channel that receives `os.Signal`'s and blocks if buffer of 10 entries is full.
I don't get what happens. Looks like `os.Interrupt` is send to `ch` to be immediately
catched below. I don't understand it - `ch` is not passed anywhere to get any other
kind of input.

Comment (stgraber) [inline lxc/exec_windows.go]: The code sets up a signal handler for os.Interrupt. If the OS sends os.Interrupt to the lxc process, Go will write that signal to the channel which will then be sent as SIGINT to LXD using the control socket.

**Comment by stgraber** [comment]
Possible, probably, easy, not so much. We'd need to somehow deploy a LXD server on Linux from the submitted branch, generate temporary credentials and then provide those credentials to appveyor for use with the client tests.

I think if we end up spending the time to hook-up testing for operating systems that can only run the LXD client (Windows and macOS), we'd probably run that in Jenkins with Windows and macOS VMs so that we can have them all sit on the same isolated network.

**Comment by brauner** [comment]
I think Jenkins is stuck.

**Comment by stgraber** [comment]
Yeah, buildd07 was unhappy, rebooted now.

**Comment by techtonik** [comment]
What's the point in having PRs with many changes? I don't like that such PRs don't carry meaningful titles in project history.

**Comment by stgraber** [comment]
Mostly laziness, it's annoying for me to maintain 3-4 branches in parallel that may need to be based on one another. The git history shows all the commits anyway so the title of the pull request really doesn't matter and instead of doing like others where they keep it titled after the first commit which is misleading, I re-title it to "bugfixes" as soon as I have more than the original fix I sent.

**Comment by techtonik** [comment]
To maintain branches I find http://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/depot_tools_tutorial.html more useful than standard `git`. It automatically rebases them and builds pretty dependency trees.

**Comment by techtonik** [comment]
Now I need to switch to your branch to get this fix for Windows, so I need to add your repo as remote. It would be slightly more convenient if PR branches were in the same repo (where master can be write protected from occassional commits).

**Comment by stgraber** [comment]
@brauner when you have a minute, this should be ready to go

**Comment by brauner** [comment]
Oh, I missed to merge. Thought I already did. :)

**Pull request merged**